toolbox/src/common.rs

```diff
@@ -352,32 +356,52 @@ pub fn transfer_creators_fee<'a, 'info>(
                     system_program,
                     currency,
                     from,
-                    from_token_acc: from_ata,
+                    from_token_acc: from_ta,
                     rent_payer,
                 } => {
-                    let creator_ata_info =
-                        unwrap_opt!(current_creator_ata_info, "missing creator ata");
-
-                    anchor_spl::associated_token::create_idempotent(CpiContext::new(
-                        associated_token_program.to_account_info(),
-                        anchor_spl::associated_token::Create {
-                            payer: rent_payer.to_account_info(),
-                            associated_token: creator_ata_info.to_account_info(),
-                            authority: current_creator_info.to_account_info(),
-                            mint: currency.to_account_info(),
-                            system_program: system_program.to_account_info(),
-                            token_program: token_program.to_account_info(),
-                        },
-                    ))?;
+                    let creator_ta_info =
+                        unwrap_opt!(current_creator_ta_info, "missing creator ata");
+
+                    // Creators can change the owner of their ATA to someone else, causing the instruction calling this
+                    // function to fail.
+                    // To prevent this, we don't idempotently create the ATA. Instead we check if the passed in token
+                    // account exists, and if it is the correct mint and owner, otherwise we create the ATA.
+
+                    if creator_ta_info.data_is_empty()
+                        && creator_ta_info.owner == &system_program::ID
+                    {
+                        anchor_spl::associated_token::create(CpiContext::new(
+                            associated_token_program.to_account_info(),
+                            anchor_spl::associated_token::Create {
+                                payer: rent_payer.to_account_info(),
+                                associated_token: creator_ta_info.to_account_info(),
+                                authority: current_creator_info.to_account_info(),
+                                mint: currency.to_account_info(),
+                                system_program: system_program.to_account_info(),
+                                token_program: token_program.to_account_info(),
+                            },
+                        ))?;
+                    } else {
+                        // Validate the mint and owner.
+                        let creator_ta =
+                            TokenAccount::try_deserialize(&mut &creator_ta_info.data.borrow()[..])?;
```

Comment:
Does this check that the account is owned by token program?

Reply:
No, but the transfer will fail if it's not owned by a token program. However, explicit checks are better than implicit so I added an account owner check here.